gfx.h

```diff
@@ -6303,13 +6303,13 @@ class GfxInternal
             "    if(isSRGB)\r\n"
             "        return ";
         mip_program_source += channel_type_str;
-        mip_program_source += '(';
+        mip_program_source += "(select(";
         mip_program_source += val_string;
-        mip_program_source += " < 0.04045f ? ";
+        mip_program_source += " < 0.04045f, ";
         mip_program_source += val_string;
-        mip_program_source += " / 12.92f : pow((";
+        mip_program_source += " / 12.92f, pow((";
         mip_program_source += val_string;
-        mip_program_source += " + 0.055f) / 1.055f, 2.4f)";
+        mip_program_source += " + 0.055f) / 1.055f, 2.4f))";
```

Comment:
Mmh, this code now using the new `select()` intrinsic will now fail if HLSL 2021 isn't supported, right?

Reply:
Yep its only 2021 or newer. Line 7161 below enables HLSL2021 as standard and gfx bundles a compatible version of dxc (which is the only thing that touches the HLSL2021 changes as the DXIR passed to the driver hasnt changed) so it would never fail aslong as everything was still within gfx (or something using gfx)

Reply:
So the only breaking change would be for any downstream user of gfx that has there own shaders that they are using gfx to compile but are not compatible with the new ```select``` requirement. However, the shader compilation output log will clearly state the need for using ```select``` and where it is required, it just unfortunately will require any user that falls into that category to make the required changes.

Reply:
I added the pre-HLSL2021 select function to the mip shader code just in case.

Reply:
Ok perfect, thanks!